### PR TITLE
CRXXX: CEN additions to the new VDV736 structures in SX

### PR DESCRIPTION
**Pull request by prirey**

### CODE REVIEW

**Review by hennoit — APPROVED** [review]

**Comment by hennoit** [comment]
To be completely consistent the element <Remark> in <RemarkContentStructure> should probably be called <RemarkText>.

**Comment by prirey** [comment]
Hi guys, thanks for merging! We are coming closer to a release. Some questions still: 
(1) I agree with the suggestion to rename "Remark". 
(2) I'm unsure about the use of Consequences in PassengerInformationAction. 
Do you pretend to have the consequence elements copied from the situation level to the passenger information action? 
What about introducing in the PtConsequenceStructure a ConsequenceRef so that we could reference to in an action?
(3) Aren't there situations where multiple advices are possible? How about a situation with replacement bus for releasing passenger jam although there is also a train as replacement during a disruption. 
(4) The element classification seems were helpful. I believe that a classification element applies for a whole publishing action and not only for a passenger information action. 
The publishing action bundles for a given scope all passenger information actions. 
Each passenger information action bundles information for different perspectives. 
Therefore the consequences for a given scope are in my opinion all the same for a publishing action. 
Do you agree?
image
(5) I was expecting that we want to rename *Name to *Text e.g. ReasonName in the ReasonGroup. Do you want to avoid renaming for compability issues?
(6) Have you thought of using in AdviceName / Details in PtAdviceStructure DefaultedTextStructure instead of NaturalLanguageStringStruc ? This also applies for ReasonName in ReasonGroup. It would be more consistent with other textual fields like Summary and SummaryText.

**Comment by hennoit** [comment]
> (5) I was expecting that we want to rename *Name to *Text e.g. ReasonName in the ReasonGroup. Do you want to avoid renaming for compability issues?
> (6) Have you thought of using in AdviceName / Details in PtAdviceStructure DefaultedTextStructure instead of NaturalLanguageStringStruc ? This also applies for ReasonName in ReasonGroup. It would be more consistent with other textual fields like Summary and SummaryText.

A quick answer to these two questions. The rest I will look into later.
In my view there is a difference in interpretation for *Text elements and *Name elements. 
- The *Text are DefaultedTextStructures, which mean that they can (in principle/default) be automatically generated from (some of the) enumerations, but you can also explicitly put a value in it if you don't have enumerations or need a better text than the default. In any case, these are usually complete sentence(s).
- The *Name are only a textual representation of / alternative for the enumeration, that can be used in the DefaultedText in stead of the enumeration. So NOT complete sentence(s) but only a (few) word(s) and therefore NaturalLanguageStructure and not DefaultedText. Different languages can still be useful here - to be used to generate DefaultedTexts in different languages.

With this reasoning the naming of ReasonName and AdviceName is correct, since they are just alternatives for TpegReason and Advice, not complete messages like SummaryText. 
AdviceDetails is an existing element that contains 'extra' information (=more details). Since it is 'extra' it probably cannot be generated and thus should not be a DefaultedText but just a NLString.

**Comment by Aurige** [comment]
Sorry to be late to the discussion (catching up from holiday time). 
May I suggest to move up the **PublicationScope**, so it can be available even with the legacy **ActionData**  ( **PublicationScope** then has to be non mandatory for backward compatibility)

image

**Comment by hennoit** [comment]
> (2) I'm unsure about the use of Consequences in PassengerInformationAction.
> Do you pretend to have the consequence elements copied from the situation level to the passenger information action?
> What about introducing in the PtConsequenceStructure a ConsequenceRef so that we could reference to in an action?

I agree that copying the complete Consequence(s) in PassengerInformationAction is not great, especially because there are also a lot of fields in it that you don't really need. An alternative is to make a new structure that contains only the elements we need. See the third 'Question' in my CR55 powerpoint on basecamp. 
Reusing the existing structures is a good idea, but in this case may be overkill, so I myself now am starting to prefer making a new simplified structure in this case.

A ConsequenceRef could also be a nice way to do it, but I think the Consequence element does not have an 'id' which we can reference here...

> (3) Aren't there situations where multiple advices are possible? How about a situation with replacement bus for releasing passenger jam although there is also a train as replacement during a disruption.

I guess multiple Advices could be possible. I just followed the current multiplicity of Advice on the situation level. If you want to have multiple Advices, then you now probably have to make multiple Consequences (each with one Advice)? 

> (4) The element classification seems were helpful. I believe that a classification element applies for a whole publishing action and not only for a passenger information action.
> The publishing action bundles for a given scope all passenger information actions.
> Each passenger information action bundles information for different perspectives.
> Therefore the consequences for a given scope are in my opinion all the same for a publishing action.
> Do you agree?

Probably the _Reason_ and parts of the _Consequence_ (namely _Condition_ and perhaps _Delay_) often will be the same for all passenger information actions belonging to a certain 'scope'. But I could imagine a different _Advice_ for each 'perspective' (= passenger information action). For example: "get out at earlier stop" for perspective=vehicleJourney and "go to replacement stop" for perspective=stopPoint.
And anyhow, in the UMS proposition there is a different ReasonText, ConsequenceText, AdviceText (etc.) for _each_ passenger information action! So in principle, if these 'defaulted texts' are not the same for all passenger information actions, then they should probably also be generated from different Reason, Consequence, Delay and Advice enumerations.

**Comment by hennoit** [comment]
> May I suggest to move up the **PublicationScope**, so it can be available even with the legacy **ActionData** ( **PublicationScope** then has to be non mandatory for backward compatibility)
> 

Putting the _PublicationScope_ outside the _PublishingAction_ does not work. The reason for having more than one _PublishingAction_ is that they have a different scope!

In the current pull request the PublicationScope has also been added to the ActionData, so it is already available for legacy stuff too:
publicationscope_in_actiondata

**Comment by Aurige** [comment]
> In the current pull request the PublicationScope has also been added to the ActionData, so it is already available for legacy stuff too:
> publicationscope_in_actiondata

Ok, fine, sorry I missed that **PublicationScope**  in **ActionData** ... that's fine, but it has to be non mandatory for backward compatibility

**Comment by hennoit** [comment]
> Thanks for your input, guys.
> 
> * I'm going to change PublicationScope to be optional for legacy issues in ActionData and Version-Att to 2.1
> * For generating standardized text, it would be nice to reused the structured information in Consequences that can be matched from the ConsequenceFilter. However, we still have to define all the parameters required for this. Therefore, I'm not sure if we are going to use the fields AdviceName/Details. I would prefere to try to use the fields AdviceRef/Type first.
>   Non-standardized generated advice text could be provided with the advice content element.
> 
> please see more details in basecamp discussion https://3.basecamp.com/3256016/buckets/9672657/messages/2949095046 👍

And my comments on basecamp too (about AdviceName and other things).

**Review by Aurige — DISMISSED** [review]

File: xsd/siri_model/siri_situationActions-v2.0.xsd [hunk 1]
```diff
@@ -181,7 +278,12 @@ Rail transport, Roads and road transport
    </xsd:element>
    <xsd:element name="Prompt" type="NaturalLanguageStringStructure" minOccurs="0" maxOccurs="unbounded">
     <xsd:annotation>
-     <xsd:documentation>Display prompt for presenting action to user.  (Unbounded since SIRI 2.0)</xsd:documentation>
+     <xsd:documentation>Display prompt for presenting action to user. (Unbounded since SIRI 2.0)</xsd:documentation>
+    </xsd:annotation>
+   </xsd:element>
+   <xsd:element name="PublicationScope" type="PublicationScopeStructure">
```
[the remark this answers is not in the document]
Comment (prirey) [reply xsd/siri_model/siri_situationActions-v2.0.xsd]: Thanks for the correction, changed!

**Review by hennoit — CHANGES_REQUESTED** [review]
In order to be consistent in the naming of elements please change _RecommendationPriority_ to _AdvicePriority_ in siri_situationActions-v2.0.xsd  (somewhere near line 640)

**Review by hennoit — APPROVED** [review]

**Review by hennoit — DISMISSED** [review]

**Comment by haeckerbaer** [comment]
See https://github.com/VDVde/UMS/pull/21, in particular the decisions made by the UmS working group that we need to discuss on december 15th.

**Comment by haeckerbaer** [comment]
Decision from UmS Meeting on december 8th:
Classification Structure is to be removed until further clarification.

**Comment by hennoit** [comment]
> Decision from UmS Meeting on december 8th:
> Classification Structure is to be removed until further clarification.

I strongly disagree (for the CEN-SIRI xsd)! 
If you have 'DefaultedTexts' in the PassengerInformationAction you should also have the values from which these default texts can be derived. These values are given in the 'Classification' structure for that specific PassengerInformationAction.
You cannot rely on the (analogous) overall classifications for this, because you do not know offhand which of those overall values are relevant for a specific PassengerInformationAction. Otherwise there also would be no reason for the texts in the PassengerInformationAction, because they would all be the same...

**Comment by haeckerbaer** [comment]
This PR includes CEN enhancements of the new VDV736v1.0 structures that were integrated into CEN SIRI v2.1.
PR is still under discussion on UmS side and won't be included in CEN SIRI v2.1.

**Pull request closed without merging**